include/zephyr/arch/arch_interface.h

```diff
@@ -1240,6 +1242,35 @@ bool arch_pcie_msi_vector_connect(msi_vector_t *vector,
  */
 void arch_spin_relax(void);
 
+#ifdef CONFIG_ARCH_STACKWALK
+/**
+ * stack_trace_consume_fn - Callback for @ref arch_stack_walk
+ * @param cookie Caller supplied pointer handed back by @ref arch_stack_walk
+ * @param addr The stack entry address to consume
+ *
+ * @return True, if the entry was consumed or skipped. False, if there is no space left to store
+ */
+typedef bool (*stack_trace_consume_fn)(void *cookie, unsigned long addr);
```

Comment:
So... I get that this is a linux API, so we can presume that it's generally well-designed and understood, yada yada.

That said... what's being "consumed" here?  When you "consume" something it goes away and isn't usable any more.   You can enumerate a dead stack frame in perpetuity, it's not going anywhere and for sure this process doesn't have side effects.   This should be a verb like "visit" or "iterate", no?  Just seems weird.

Reply:
Also: do we want to consider adding a parameter for the instruction pointer (e.g. "return_address", "calling_address", depending on convention) here?  Most architectures provide that too in some form (I guess we could allow it to be NULL if unsupported), and for sure anyone using this API will want it.

Not sure how linux provides that: is it another API you call with the stack address as a parameter maybe?

Reply:
`stack_trace_consume_fn` API is actually the same as Linux:

https://github.com/torvalds/linux/blob/cc8ed4d0a8486c7472cd72ec3c19957e509dc68c/include/linux/stacktrace.h#L11-L39

Reply:
I named it `stack_trace_callback_fn ` instead, since it is a callback